test/integration/server.h

```diff
@@ -433,6 +521,8 @@ class IntegrationTestServerImpl : public IntegrationTestServer {
   Stats::Store* stat_store_{};
   Network::Address::InstanceConstSharedPtr admin_address_;
   absl::Notification server_gone_;
+  Stats::SymbolTablePtr symbol_table_;
+  Stats::NotifyingAllocatorImpl stats_allocator_;
```

Comment:
How about plumbing a ctor param to allow integration tests to choose between production stats and suppporting waitaForCounter*?